src/GrassPokemon.java

```diff
@@ -0,0 +1,156 @@
+    }
+
+    @Override
+    public String getType() {
+        return "Grass";
+    }
+
+    @Override
+    public List<String> getAttacks() {
+        return attacks;
+    }
+
+    @Override
+    public void setHp(int damage) {
+        hp = hp - damage;
+    }
+
+    public void leafStorm(Pokemon pokemon, Pokemon gymPokemon) {
+        System.out.println(pokemon.getName()
+                + " attacks "
+                + gymPokemon.getName()
+                + " with Leaf Storm");
+        grassAttack(gymPokemon);
+    }
+
+    public void solarBeam(Pokemon pokemon, Pokemon gymPokemon) {
+        System.out.println(pokemon.getName()
+                + " attacks "
+                + gymPokemon.getName()
+                + " with Solar Beam");
+        grassAttack(gymPokemon);
+    }
+
+    public void leechSeed(Pokemon pokemon, Pokemon gymPokemon) {
+        System.out.println(pokemon.getName()
+                + " attacks "
+                + gymPokemon.getName()
+                + " with Leech Seed");
+        grassAttack(gymPokemon);
+        if(Objects.equals(gymPokemon.getType(), "Electric")){
+            pokemon.setHp(-20);
+            System.out.println(pokemon.getName() + " stole 20 hp.");
+        }
+        if(Objects.equals(gymPokemon.getType(), "Fire")){
+            pokemon.setHp(-15);
+            System.out.println(pokemon.getName() + " stole 15 hp.");
+        }
+        if(Objects.equals(gymPokemon.getType(), "Water")){
+            pokemon.setHp(-10);
+            System.out.println(pokemon.getName() + " stole 10 hp.");
+        }
+        if(Objects.equals(gymPokemon.getType(), "Grass")){
+            pokemon.setHp(-5);
+            System.out.println(pokemon.getName() + " stole 5 hp.");
+        }
+    }
+
+    public void leaveBlade(Pokemon pokemon, Pokemon gymPokemon) {
+        System.out.println(pokemon.getName()
+                + " attacks "
+                + gymPokemon.getName()
+                + " with Leave Blade");
+        grassAttack(gymPokemon);
+    }
+
+    private void grassAttack(Pokemon gymPokemon) {
+        if(Objects.equals(gymPokemon.getType(), "Electric")){
+            System.out.println("It's Super Effective!");
+            System.out.println(gymPokemon.getName()
+                    + " loses 20 hp.");
+            gymPokemon.setHp(20);
+            System.out.println(gymPokemon.getName()
+                    + " has "
+                    + gymPokemon.getHp()
+                    + " hp remaining.");
+        }
+        if(Objects.equals(gymPokemon.getType(), "Fire")){
+            System.out.println("It's Effective!");
+            System.out.println(gymPokemon.getName()
+                    + " loses 15 hp.");
+            gymPokemon.setHp(15);
+            System.out.println(gymPokemon.getName()
+                    + " has "
+                    + gymPokemon.getHp()
+                    + " hp remaining.");
+        }
+        if(Objects.equals(gymPokemon.getType(), "Water")){
```

Comment:
In plaats van `Object.equals` kun je ook `gymPokemon.getType().equals("Water")` gebruiken, maar beide is goed